docs/streaming-programming-guide.md

```diff
@@ -644,17 +644,90 @@ methods for creating DStreams from files as input sources.
     </div>
     </div>
 
-	Spark Streaming will monitor the directory `dataDirectory` and process any files created in that directory (files written in nested directories not supported). Note that
+	Spark Streaming will monitor the directory `dataDirectory` and process any files created in that directory.
+
+     ++ The files must have the same data format.
+     + A simple directory can be monitored, such as `hdfs://namenode:8040/logs/`.
+       All files directly such a path will be processed as they are discovered.
+     + A POSIX glob pattern can be supplied, such as
+       `hdfs://namenode:8040/logs/2016-??-31`.
+       Here, the DStream will consist of all files directly under those directories
+       matching the regular expression.
```

Comment:
I think this comment is still incorrect. A glob is not a regex. If this is just the syntax other Hadoop APIs support, that seems reasonable, but it should be described that way.